lib/sbuf.c

```diff
@@ -63,13 +63,12 @@ void sbuf_push(struct sbuf *buf, int indent, const char *format, ...)
 	int written;
 
 	if (!buf->fixed) {
-		char dummy;
 		int written1, written2;
 		size_t new_size;
 
-		written1 = snprintf(&dummy, 0, "%*s", indent, "");
+		written1 = snprintf(NULL, 0, "%*s", indent, "");
```

Comment:
Isn't this expression equivalent to the value of `indent`?

Reply:
I'm not sure I'm understanding you, Quentin. The purpose of snprintf with 0-size is to simulate the operation in order to get the size so the buffer could be allocated afterward. The change is because using the "dummy" storage is unnecessary, and because of the Coverity complaint :-)

Reply:
Yes, `snprintf` tells you how many bytes would have been written if there were enough space. But since you're using `"%*s"` with an empty string you just get `indent` number of spaces. Hence my comment that the expression `snprintf(NULL, 0, "%*s", indent, "")` is equivalent to the value of `indent`, i.e. you could just do
```
written1 = indent;
```

![screen shot 2018-06-21 at 3 17 58 pm](https://user-images.githubusercontent.com/6827003/41740644-a0e693f6-7566-11e8-92c1-6828385c662f.png)